SwiftUIBasics/Views/SignUpView.swift

```diff
@@ -52,6 +76,50 @@ class SignUpViewModel: ObservableObject {
             }
             .assign(to: \.isValidPasswordUpperCase, on: self)
             .store(in: &cancelableSet)
+        
+        // isValidPasswordLowerCase
+        $password
+            .receive(on: RunLoop.main)
+            .map { password in
+                let pattern = "[a-z]"
+                if let _ = password.range(of: pattern, options: .regularExpression) {
+                    return true
+                } else {
+                    return false
+                }
+            }
+            .assign(to: \.isValidPasswordLowerCase, on: self)
+            .store(in: &cancelableSet)
+
+        // isValidPasswordHasASymbol
+        $password
+            .receive(on: RunLoop.main)
+            .map { password in
+                let pattern = "[!@#$%^&*()_+\\-=\\[\\]{};':\"\\\\|,.<>\\/?]"
```

Comment:
for symbols use the non word regex 
```suggestion
                let pattern = "[\\W]"
```